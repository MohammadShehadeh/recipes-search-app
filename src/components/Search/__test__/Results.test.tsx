/* eslint-disable jsx-a11y/anchor-is-valid */
import renderer from 'react-test-renderer';
import { Results } from '..';

const recipesMock = [
  {
    image: 'image',
    id: 10,
    title: 'title',
  },
];

jest.mock('react-router-dom', () => ({
  ...(jest.requireActual('react-router-dom') as any),
  Link: ({ children }: any) => <a data-mocked-link>{children}</a>,
}));

describe('Results', () => {
  it('should render "Loading" to match snapshot', () => {
    const tree = renderer.create(<Results isLoading />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render "Results" to match snapshot', () => {
    const tree = renderer.create(<Results recipes={recipesMock} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
