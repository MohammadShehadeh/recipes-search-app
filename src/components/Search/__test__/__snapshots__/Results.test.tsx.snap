// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results should render "Loading" to match snapshot 1`] = `
<div
  className="grid"
>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <div
      className="skeleton"
    >
      <div
        className="loading"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <div
      className="skeleton"
    >
      <div
        className="loading"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <div
      className="skeleton"
    >
      <div
        className="loading"
        style={Object {}}
      />
    </div>
  </div>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <div
      className="skeleton"
    >
      <div
        className="loading"
        style={Object {}}
      />
    </div>
  </div>
</div>
`;

exports[`Results should render "Results" to match snapshot 1`] = `
<div
  className="grid"
>
  <div
    className="col col-sm-12 col-md-6 col-lg-4"
  >
    <a
      data-mocked-link={true}
    >
      <h3
        className="title"
      >
        title
      </h3>
      <div
        className="image"
      >
        <img
          alt="title"
          src="image"
        />
      </div>
    </a>
  </div>
</div>
`;
